File: server/models/OperationLog.js
const mongoose = require('mongoose');

const operationLogSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  username: {
    type: String,
    required: true
  },
  module: {
    type: String,
    enum: ['products', 'packingLists', 'users', 'system'],
    required: true
  },
  action: {
    type: String,
    enum: ['create', 'update', 'delete', 'read', 'import', 'export', 'backup'],
    required: true
  },
  description: {
    type: String,
    required: true
  },
  details: {
    type: mongoose.Schema.Types.Mixed
  },
  ip: String,
  userAgent: String,
  status: {
    type: String,
    enum: ['success', 'failure'],
    default: 'success'
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

// 创建索引
operationLogSchema.index({ createdAt: -1 });
operationLogSchema.index({ userId: 1, createdAt: -1 });
operationLogSchema.index({ module: 1, action: 1, createdAt: -1 });

module.exports = mongoose.model('OperationLog', operationLogSchema); 

File: backend/app/models/operation_log.py
from sqlalchemy import Column, String, Enum, JSON, ForeignKey
from sqlalchemy.orm import relationship
from .base import BaseModel
import enum

class LogModule(str, enum.Enum):
    """日志模块"""
    PRODUCTS = "products"
    PACKING_LISTS = "packing_lists"
    USERS = "users"
    SYSTEM = "system"

class LogAction(str, enum.Enum):
    """日志操作类型"""
    CREATE = "create"
    UPDATE = "update"
    DELETE = "delete"
    READ = "read"
    IMPORT = "import"
    EXPORT = "export"
    BACKUP = "backup"

class LogStatus(str, enum.Enum):
    """日志状态"""
    SUCCESS = "success"
    FAILURE = "failure"

class OperationLog(BaseModel):
    """操作日志模型"""
    __tablename__ = "operation_logs"

    user_id = Column(ForeignKey("users.id"), nullable=True)
    username = Column(String, nullable=False)
    module = Column(Enum(LogModule), nullable=False)
    action = Column(Enum(LogAction), nullable=False)
    description = Column(String, nullable=False)
    details = Column(JSON, nullable=True)
    ip = Column(String, nullable=True)
    user_agent = Column(String, nullable=True)
    status = Column(Enum(LogStatus), default=LogStatus.SUCCESS)

    # 关联
    user = relationship("User") 